### ISSUE TITLE
Where is the File module Final Report?

### ISSUE THREAD

**Issue by fgeorges** [expath/expath-cg#128]
@ChristianGruen, the wiki <https://www.w3.org/community/expath/wiki/Modules> mentioned the File module as being a Final Report.  But this is not the case on the CG homepage <https://www.w3.org/community/expath/>.

If this is "only" a process thing (if it was actually granted the final status, but not published properly on the CG page), I can ask for publication quite quickly.

We might as well take this opportunity to switch it to ReSpec before (only cosmetic changes).

What do you think?

**Comment by ChristianGruen** [comment]
Sounds good! What’s the fastest way to get this done?

**Comment by fgeorges** [comment]
I can initiate the process with W3C staff, it is still fresh in my mind :)  Do you want to switch it to ReSpec first?  It would be good to align our Final Reports I think.

**Comment by ChristianGruen** [comment]
I’m not sure when I’ll have time to convert the spec to ReSpec. Maybe it’s less effort if we stick to the current format?

**Comment by michaelhkay** [comment]
I'm not quite sure what the current format is, but for specs that are essentially lists of function specs, I find it really useful to have the source specs in F&O function-catalog format, so it's possible for tools to extract the function signatures, argument names, etc. We shouldn't do anything that reduces the semantic richness of existing markup.

Michael Kay
Saxonica

**Comment by johnlumley** [comment]
I might possibly have an XSLT conversion of the File Spec into F&O, at least for the catalog. I'll have a look later.

John

**Comment by michaelhkay** [comment]
I see. So (if I understand correctly) we did it in reverse - extracted the function-catalog.xml by up-conversion from the spec prose, rather than generating the spec prose from the function catalog.

Mike

**Comment by johnlumley** [comment]
> On 19 Nov 2018, at 14:57, Michael Kay <notifications@github.com> wrote:
> 
> I see. So (if I understand correctly) we did it in reverse - extracted the function-catalog.xml by up-conversion from the spec prose, rather than generating the spec prose from the function catalog.
> 
> Mike
> 


Yes - was the easiest way..... In constructing the Binary spec we did indeed progress catalog -> spec, but the File spec had been written in full prose by Christian et al sometime before mid-2013.  It was fairly simple to extract what we needed for the catalog to drive the requirements of Saxon documentation generation.

John

**Comment by ChristianGruen** [comment]
Latest version: https://qt4cg.org/specifications/expath-file-40/Overview.html
